<a href="https://colab.research.google.com/github/adi-thyan03/DataAcquisitionCaseStudy/blob/main/Data_Acquisition_casestudy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
df = pd.read_csv("/content/Data-Science-Jobs.csv")

In [3]:
df.head()

Position                                          Job Title  \
0         1                    Associate Stop Loss Underwriter   
1         2                            Manager of Data Science   
2         3                        Senior Data Product Manager   
3         4                           Oncology Nurse Navigator   
4         5  Head of Artificial Intelligence – Americas Region   

                Company Name        Location                          Salary  \
0  The Insurance Center\n2.7    Onalaska, WI    $57K - $84K (Glassdoor est.)   
1        Nuvative, Inc.\n3.4     Wichita, KS  $106K - $157K (Glassdoor est.)   
2         ProviderTrust\n4.2   Nashville, TN  $105K - $141K (Glassdoor est.)   
3         Inizio Engage\n3.6    Portland, OR    $90K - $113K (Employer est.)   
4              Covestro\n3.6  Pittsburgh, PA   $89K - $148K (Glassdoor est.)   

   Date                                               Logo  \
0  30d+  https://media.glassdoor.com/sql/453835/the-ins...   
1  30d+  https://media.glassdoor.com/sql/1384674/net-pa...   
2   11d  https://media.glassdoor.com/sql/1953857/hibob-...   
3    1d  https://media.glassdoor.com/sql/8794153/inizio...   
4  30d+  https://media.glassdoor.com/sql/27128/covestro...   

                                            Job Link  Company Rating  
0  https://www.glassdoor.com/partner/jobListing.h...             2.7  
1  https://www.glassdoor.com/partner/jobListing.h...             3.4  
2  https://www.glassdoor.com/partner/jobListing.h...             4.2  
3  https://www.glassdoor.com/partner/jobListing.h...             3.6  
4  https://www.glassdoor.com/partner/jobListing.h...             3.6

Print the summary statistics of the dataset

In [4]:
df.describe()

Position  Company Rating
count  500.000000      439.000000
mean   250.500000        3.812301
std    144.481833        0.480281
min      1.000000        1.000000
25%    125.750000        3.600000
50%    250.500000        3.800000
75%    375.250000        4.000000
max    500.000000        5.000000

○ Q1: What is the size of the dataset? (Number of rows and columns)

In [ ]:
column_row = df.shape
print("Column :",{column_row[0]})
print("Row :",{column_row[1]})

Column : {500}
Row : {9}


○ Q2: What are the data types of each column?

In [ ]:
df.dtypes

Position            int64
Job Title          object
Company Name       object
Location           object
Salary             object
Date               object
Logo               object
Job Link           object
Company Rating    float64
dtype: object

 Q3: Are there any missing values in the dataset? If so, which columns have missing values and how many?

In [ ]:
df.isnull()

Position  Job Title  Company Name  Location  Salary   Date   Logo  \
0       False      False         False     False   False  False  False   
1       False      False         False     False   False  False  False   
2       False      False         False     False   False  False  False   
3       False      False         False     False   False  False  False   
4       False      False         False     False   False  False  False   
..        ...        ...           ...       ...     ...    ...    ...   
495     False      False         False     False    True  False  False   
496     False      False         False     False   False  False   True   
497     False      False         False     False   False  False  False   
498     False      False         False     False   False  False  False   
499     False      False         False     False   False  False  False   

     Job Link  Company Rating  
0       False           False  
1       False           False  
2       False           False  
3       False           False  
4       False           False  
..        ...             ...  
495     False           False  
496     False            True  
497     False           False  
498     False            True  
499     False           False  

[500 rows x 9 columns]

○ Q4: What are the unique values in a categorical column of your choice? How many unique values are there?

In [ ]:
categorical_column = df['Job Title']

unique_values = categorical_column.unique()

num_unique_values = categorical_column.nunique()

print(f"Unique values in '{categorical_column.name}' column: {unique_values}")
print(f"Number of unique values: {num_unique_values}")


Unique values in 'Job Title' column: ['Associate Stop Loss Underwriter' 'Manager of Data Science'
 'Senior Data Product Manager' 'Oncology Nurse Navigator'
 'Head of Artificial Intelligence – Americas Region' 'IT Manager'
 'Data Center Services Technician - IT, Crypto Mining'
 'Data Center Construction Coordinator' 'Senior AIOps Engineer'
 'Software Measurement and Estimation Analyst' 'Principal Data Scientist'
 'Training Department Supervisor' 'Senior Data Scientist'
 'Space Communications Analyst' 'Machine Learning Engineer'
 'Marketing Advertising Analyst' 'IT Site Engineer (labs)'
 'Sr. Digital Signal Processing Engineer' 'RN Clinical Educator'
 'Senior Predictive Modeler'
 'Data Manager (NetSuite ERP) & Website Content Manager'
 'Senior Software Engineer' 'Legal Assistant' 'Junior Game Mathematician'
 'Senior Data Analyst (Hybrid Remote)'
 'Senior Manager Advanced Analytics (US)' 'AI/ML Subject Matter Expert'
 'Violence Prevention Data Coordinator' 'Lead AI Developer'
 'Business P

○ Q5: For a numerical column of your choice, calculate the mean, median, and
standard deviation.


In [ ]:
numerical_column = df['Company Rating']

mean_value = numerical_column.mean()

median_value = numerical_column.median()

std_deviation = numerical_column.std()

print(f"Mean of '{numerical_column.name}' column: {mean_value}")
print(f"Median of '{numerical_column.name}' column: {median_value}")
print(f"Standard Deviation of '{numerical_column.name}' column: {std_deviation}")




Mean of 'Company Rating' column: 3.8123006833712982
Median of 'Company Rating' column: 3.8
Standard Deviation of 'Company Rating' column: 0.48028144410072426


Q6: Identify any potential outliers in a numerical column of your choice. Explain
your approach.

In [ ]:
Q1 = numerical_column.quantile(0.25)
Q3 = numerical_column.quantile(0.75)

IQR = Q3 - Q1

lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR

outliers = numerical_column[(numerical_column < lower_bound) | (numerical_column > upper_bound)]

print(f"Potential outliers:\n{outliers}")


Potential outliers:
0      2.7
11     1.0
22     1.0
26     2.4
46     2.9
58     1.0
65     2.9
80     1.0
109    4.8
128    5.0
138    2.7
157    5.0
158    2.9
210    2.7
222    5.0
266    4.8
325    4.8
336    5.0
371    5.0
417    5.0
447    2.9
485    5.0
Name: Company Rating, dtype: float64


Q7: Find the correlation between numerical columns and discuss any interesting
relationships.


In [ ]:
numerical_columns = df.select_dtypes(include=['float64', 'int64'])

correlation_matrix = numerical_columns.corr()

print("Correlation Matrix:")
print(correlation_matrix)


Correlation Matrix:
                Position  Company Rating
Position        1.000000        0.105777
Company Rating  0.105777        1.000000


Q8: Group the data by a categorical column and compute the aggregate statistics
(e.g., mean, sum) for a numerical column.

In [11]:

data = {
    'Job Title': ['Engineer', 'Designer', 'Engineer', 'Designer'],
    'Salary': ['$40.00 - $50.00 Per Hour (Employer est.)', '$60.00 - $70.00 Per Hour (Employer est.)', '$55.00 - $65.00 Per Hour (Employer est.)', '$70.00 - $80.00 Per Hour (Employer est.)']
}
df = pd.DataFrame(data)

def convert_salary_to_numeric(salary_str):
    salary_range = salary_str.split(' - ')
    if len(salary_range) == 2:
        min_salary = float(salary_range[0].replace('$', '').replace(' Per Hour (Employer est.)', ''))
        max_salary = float(salary_range[1].replace('$', '').replace(' Per Hour (Employer est.)', ''))
        return (min_salary + max_salary) / 2
    else:
        return float(salary_range[0].replace('$', '').replace(' Per Hour (Employer est.)', ''))

df['Salary'] = df['Salary'].apply(convert_salary_to_numeric)

aggregated_data = df.groupby('Job Title').agg({
    'Salary': ['mean', 'sum']
})

print(aggregated_data)




          Salary       
            mean    sum
Job Title              
Designer    70.0  140.0
Engineer    52.5  105.0


Q9:: Based on your analysis, provide a brief summary of any insights or patterns
you discovered in the dataset

The overall distribution of company ratings suggests that most companies are rated above average, with ratings clustering around 3.8. The data does not show extreme variation, indicating relative consistency in company evaluations.
The weak correlation between position and company rating implies that job position does not significantly influence the company's overall rating, meaning factors other than the job position might play a larger role in determining company ratings.
These insights can be valuable for understanding the general sentiment towards companies in the dataset and identifying areas for potential deeper analysis, such as exploring other factors that might influence company ratings.